app/controllers/api/base_controller.rb

```diff
@@ -39,8 +39,8 @@ class BaseController < ActionController::Base
 
     rescue_from Foreman::MaintenanceException, :with => :service_unavailable
 
-    def get_resource
-      instance_variable_get(:"@#{resource_name}") || raise('no resource loaded')
+    def get_resource(options = {})
```

Comment:
I guess this would be a question the whole @theforeman/core, probably on mailing list? Personally +1, but I would be interested into opinions of wider range before going for it